### PR TITLE
Clean-up sample live tests

### PR DESCRIPTION
**Pull request by weshaggard**
- Remove unnecessary PS module importing
- Remove dead code in post script
- Add retry/sleep to ensure the iothub is active before trying to make changes to it
- Remove unnecessary docker image for ubuntu
- Remove unused and outdated Generate-X509-Certificate.ps1 script.

### CODE REVIEW

**Review by vhvb1989 — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Is the intermittent macOS failure expected?
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=845422&view=logs&j=4a578679-1661-5a93-5c59-ed624155db79&t=31426749-be43-5a30-23a5-7dba65be00b3&s=4ff3786a-ea67-5356-a1c6-e7ae13aeb7e1

**Comment by weshaggard** [comment]
That is a good question. The failure actually looks like it is an issue with the IotHub still not being available yet. I explicitly added the State check in a retry loop to avoid this and that seems to have been working in all my other testing with it. However perhaps there is still some odd IotHub deployment delays. I retriggered to see if it gets hit again but we might need to figure out another way to correctly query the state of the IotHub or maybe implement a retry logic for the first device we hook up.

**Comment by weshaggard** [comment]
It passed on retry but I'm going to do some more runs to see if it hits again.

**Review by danewalton — APPROVED** [review]
thanks for the help here Wes

File: sdk/samples/iot/templates/iothub/test-resources-post.ps1 [hunk 1]
```diff
@@ -56,6 +37,20 @@ $fingerprint = Get-Content -Path .\fingerprint.txt
 # Get the hub as an object
 $hub_obj = Get-AzIotHub -ResourceGroupName $ResourceGroupName -Name $iothubName
 
+$retryCount = 1
+while ($retryCount -lt 5 -and $hub_obj.Properties.State -ne "Active")
+{
+  Write-Host "AzIotHub is not yet active so sleeping for $retryCount seconds"
+  Start-Sleep -Seconds $retryCount
```
Comment (ahsonkhan) [inline sdk/samples/iot/templates/iothub/test-resources-post.ps1:44]: Nice comment and logic. Fun linear back-off time for the retry (1s, then 2s, ...). Is 10 second good enough as the max timeout?

File: sdk/samples/iot/templates/iothub/test-resources-post.ps1 [hunk 2]
```diff
@@ -56,6 +37,20 @@ $fingerprint = Get-Content -Path .\fingerprint.txt
...
+  $hub_obj = Get-AzIotHub -ResourceGroupName $ResourceGroupName -Name $iothubName
+  $retryCount++
+}
+
+if ($hub_obj.Properties.State -ne "Active")
+{
+  Write-Warning "AzIotHub instances is not yet active so later steps might fail as a result. If they fail often you can increase the retry count above."
```
Comment (ahsonkhan) [inline sdk/samples/iot/templates/iothub/test-resources-post.ps1:51]: Actionable fix. Great!

**Review by ahsonkhan — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Aww, failed again, on linux this time.

**Comment by weshaggard** [comment]
>Aww, failed again, on linux this time.

Both times I've seen this fail in my testing it didn't even hit the retry logic so there must still be a slight race condition. So I'm going to tweak the logic a little such that I always sleep once through the loop to give it a little time automatically.

File: sdk/samples/iot/templates/iothub/test-resources-post.ps1 (hunk 1 above)
Comment (weshaggard) [reply sdk/samples/iot/templates/iothub/test-resources-post.ps1:42]: I didn't want to always blindly sleep for a fixed amount of time. I wanted to try and continue as quickly as possible.

File: sdk/samples/iot/templates/iothub/test-resources-post.ps1 [hunk 3]
```diff
@@ -17,44 +17,39 @@ param (
...
 openssl x509 -noout -fingerprint -in device_ec_cert.pem | % {$_.replace(":", "")} | % {$_.replace("SHA1 Fingerprint=", "")} | Tee-Object -FilePath fingerprint.txt
 $fingerprint = Get-Content -Path .\fingerprint.txt
 
-# Get the hub as an object
-$hub_obj = Get-AzIotHub -ResourceGroupName $ResourceGroupName -Name $iothubName
+$retryCount = 0
+do
+{
+  $retryCount++
+  Write-Host "AzIotHub is not yet active so sleeping for $retryCount seconds"
```
Comment (ahsonkhan) [inline sdk/samples/iot/templates/iothub/test-resources-post.ps1:41]: nit:
```suggestion
  Write-Host "AzIotHub is not yet active so sleeping for $retryCount seconds."
```

**Pull request merged**